### DISCUSSION TITLE
Compile and debug FDS in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by lihy213** [firemodels/fds#13021] [Q&A]
Dear developers,

I want to develop FDS in my way and debug it in VS 2022. I copy all the source code files from the Source file to the VS project file, and the compilation reports errors such as “*error #7002: Error in opening the compiled module file. check INCLUDE paths. [MPI_F08]*”. But I did add the mpi paths(e.g. *C:\Program Files (x86)\Intel\oneAPI\mpi\latest\include*) from the HPC installation to the project file and still got the same error. How can I set up the environment in VS to be able to debug?

Kind regards,

LI

**Comment by lihy213** [comment]
Well, I added the mpi paths from HPC to the project property correctly, selected the `fpp` scheme for preprocessor, and was able to debug successfully. However, the generated executable file (e.g. my file name is `fds_test.exe`) cannot be used to compute the fds cases, and when I run `fds_test.exe` on its own, it reports the error “*Unable to locate the program input point _for_ieee_quiet_eq_k8_ in the dynamic link library fds_test. exe*", what is the reason for this? Are there some problems with one of the step settings?
![image](https://github.com/firemodels/fds/assets/96673442/7b6410e3-ee03-4ea5-ad0d-6f42b3037d61)
![image](https://github.com/firemodels/fds/assets/96673442/ecb12cde-1e63-45e9-a8f1-3454edb733f6)

**Comment by drjfloyd** [comment]
I don't know that any of the dev team builds FDS from inside VS.  Clone the entire FDS repo with git (make sure git is in your path), install make (like the make from gnuwin32 and make sure it is in your path), then cd into the Build directory into the FDS repo and then into the version you want to build, and run the make_fds.bat file in the version folder.

More detailed instructions here:
https://github.com/firemodels/fds/wiki/Git-Notes-Getting-Started
https://github.com/firemodels/fds/wiki/FDS-Compilation
https://github.com/firemodels/fds/wiki/Compiling-FDS-with-Intel-oneAPI

**Comment by lihy213** [comment]
It has been solved!
